server/repository/src/db_diesel/clinician.rs

```diff
@@ -0,0 +1,300 @@
+                    apply_sort_no_case!(query, sort, clinician_dsl::last_name)
+                }
+                ClinicianSortField::Initials => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::initials)
+                }
+                ClinicianSortField::RegistrationCode => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::registration_code)
+                }
+                ClinicianSortField::Category => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::category)
+                }
+                ClinicianSortField::Address1 => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::address1)
+                }
+                ClinicianSortField::Address2 => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::address2)
+                }
+                ClinicianSortField::Phone => apply_sort_no_case!(query, sort, clinician_dsl::phone),
+                ClinicianSortField::Mobile => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::mobile)
+                }
+                ClinicianSortField::Email => apply_sort_no_case!(query, sort, clinician_dsl::email),
+                ClinicianSortField::Female => {
+                    apply_sort_no_case!(query, sort, clinician_dsl::is_female)
+                }
+            }
+        } else {
+            query = query.order(clinician_dsl::id.asc())
+        }
+
+        let final_query = query
+            .offset(pagination.offset as i64)
+            .limit(pagination.limit as i64);
+
+        // Debug diesel query
+        // println!(
+        //     "{}",
+        //     diesel::debug_query::<DBType, _>(&final_query).to_string()
+        // );
+
+        let result =
+            final_query.load::<ClinicianAndClinicianStoreJoin>(&self.connection.connection)?;
+
+        Ok(result.into_iter().map(to_domain).collect())
+    }
+}
+
+fn to_domain(
+    (clinician_row, clinician_store_join_row): ClinicianAndClinicianStoreJoin,
+) -> Clinician {
+    Clinician {
+        clinician_row,
+        clinician_store_join_row,
+    }
+}
+
+// clinician_store_join_dsl::clinician_id.eq(clinician_dsl::id)
+type ClinicianIdEqualToId = Eq<clinician_store_join_dsl::clinician_id, clinician_dsl::id>;
+// clinician_store_join_dsl::store_id.eq(store_id)
+type StoreIdEqualToStr = Eq<clinician_store_join_dsl::store_id, String>;
+// clinician_store_join_dsl::clinician_store_join.on(clinician_id.eq(clinician_dsl::id))
+type OnClinicianStoreJoinToClinicianJoin =
+    OnClauseWrapper<clinician_store_join::table, And<ClinicianIdEqualToId, StoreIdEqualToStr>>;
+
+type BoxedClinicianQuery =
+    IntoBoxed<'static, LeftJoin<clinician::table, OnClinicianStoreJoinToClinicianJoin>, DBType>;
+
+fn create_filtered_query(store_id: String, filter: Option<ClinicianFilter>) -> BoxedClinicianQuery {
+    let mut query = clinician_dsl::clinician
+        .left_join(
```

Comment:
technically this sort of store join is not needed for clinician (it's needed for name <-> name_store_join since name_store_join has other properties we want to use in the filter). Also are we using name_store_join somewhere ? If not this join has no affect.

I think I would just do is_visible filter, with inner query for clinician_store_join matching store_id

Reply:
thanks! I simplified it and added a test